feat(guidelines): add new guideline for money fields (#89)
otto-de/api-guidelines: decisions/0001-should-use-common-otto-decimal-format.md

# [0001] New API guideline rule for `otto:decimal` format

- Status: `proposed`
- Decided by: <jens.fischer@otto.de>, <max.edenharter@otto.de>
- Date: 2025-05-19

## Context

> We want to introduce a new API guideline rule for defining the data type and format for higher precision decimals as used e.g. in calculation with monetary amounts. This rule will be part of the JSON chapter applicable for both synchronous REST APIs and asynchronous Events.

Reviewer: ```suggestion
Using a unified format for decimals supports precision when relying on default conversion from `number` to `double` or `float`, as described in the [Zalando API guidelines][zalando-notes].
Implementing a new API guideline to define the data type and format can help with higher precision in decimals as used, for example, in calculation with monetary amounts.
This rule will be part of the JSON chapter applicable for both synchronous REST APIs and asynchronous events.
```

> Not having such a format could lead to lose precision when reyling on default conversion from `number` to `double` or `float` as described at [Zalando's API guidelines][zalando-notes].

Reviewer: ```suggestion
```

## Options

### Option 1

> Keep using formats `double` or `float` and accept that naive implementations implicitly accepting the default conversion run at risk to introduce calculation errors.

Reviewer: ```suggestion
Keep using formats `double` or `float` and accept that naive implementations that implicitly accept the default conversion run at risk of introducing calculation errors.
```

### Option 2

> Use a custom format `otto:decimal` that forces implementation to make informed decisions how to convert it.

Reviewer: ```suggestion
Use a custom format `otto:decimal` that enforces making informed decisions on how to convert it during implementation.
```

## Decision

> Introduce [new API guideline rule][rule-R100079]

Reviewer: ```suggestion
Introduce [new API guideline rule][rule-R100079].
```

# [0001] Use `otto:decimal` format for higher precision in amounts specified in APIs

- Status: `accepted`
- Decided by: <jens.fischer@otto.de>, <max.edenharter@otto.de>
- Date: 2025-07-31

## Context

Using a unified format for decimals supports precision when relying on default conversion from `number` to `double` or `float`, as described in the [Zalando API guidelines][zalando-notes].
Implementing a new API guideline to define the data type and format can help with higher precision in decimals as used, for example, in calculation with monetary amounts.
This rule will be part of the JSON chapter applicable for both synchronous REST APIs and asynchronous events.

Not having such a unified format could lead to losing precision when relying on default conversion from `number` to `double` or `float`.

## Options

### Option 1

Keep using formats `double` or `float` and accept that naive implementations that implicitly accept the default conversion run at risk of introducing calculation errors.

### Option 2

> Use a custom format `otto:decimal` that enforces making informed decisions on how to convert it during implementation.

Reviewer: ```suggestion
Use a custom format `otto:decimal` that enforces making informed decisions on how to convert from `number` to `double` or `float` during implementation.
```
Alternatively, `how to convert decimals`.
